RandBLAS/trig_skops.hh

```diff
@@ -0,0 +1,493 @@
+#include "RandBLAS/base.hh"
+#include "RandBLAS/exceptions.hh"
+#include "RandBLAS/random_gen.hh"
+#include <RandBLAS/sparse_skops.hh>
+
+#include <Random123/philox.h>
+#include <blas.hh>
+#include <lapack.hh>
+#include <omp.h>
+
+#include <iostream>
+#include <stdio.h>
+#include <stdexcept>
+#include <string>
+#include <tuple>
+#include <random>
+
+#include <math.h>
+#include <typeinfo>
+
+#define MAX(a, b) (((a) < (b)) ? (b) : (a))
+#define MIN(a, b) (((a) < (b)) ? (a) : (b))
+
+namespace RandBLAS {
+
+    // =============================================================================
+    /// WARNING: None of the following functions or overloads thereof are part of the
+    /// public API
+    ///
+    template<SignedInteger sint_t>
+    void generateRademacherVector_r123(sint_t* buff, uint32_t key_seed, uint32_t ctr_seed, int64_t n) {
+        typedef r123::Philox2x32 RNG;
+
+        // Use OpenMP to parallelize the Rademacher vector generation
+        #pragma omp parallel
+        {
+            // Each thread has its own RNG instance
+            RNG rng;
+
+            RNG::key_type k = {{key_seed + omp_get_thread_num()}}; // Unique key for each thread
```

Comment:
1. This function should take in an RNGState and return an RNGState.
2. Parallel generation should not rely on different keys. In the situations where we need parallelism we make a point to ensure that the number of threads used doesn't affect the outcome of the random number generation.
3. I'm doubtful that parallelism is useful here. We use parallelism when generating many vectors at once. I have a preference for _no OpenMP here_ unless you can show there's a compelling performance reason to keep it.

Reply:
Added an overload of the generateRademacherVector_r123 method with the appropriate signature and removed the unnecessary OpenMP dependency (the performance gain in the case of `std::random` was very significant when switching to a parallel implementation but not so much in the case of Random123)